File: appunti/algoritmi/ricerca/codice/ricerca_binaria_iterativa.js
const NON_PRESENTE = Symbol("non presente");

const ricerca_binaria = function(A, n, v) {
  let sinistra = 0,
    destra = n - 1;
  while (sinistra <= destra) {
    let somma = sinistra + destra;
    let centro = somma % 2 === 0 ? somma / 2 : (somma - 1) / 2;
    if (A[centro] > v) {
      destra = centro - 1;
    } else if (A[centro] < v) {
      sinistra = centro + 1;
    } else {
      return centro;
    }
  }

  return NON_PRESENTE;
};

File: appunti/algoritmi/ricerca/codice/ricerca_binaria_iterativa.py
NON_PRESENTE = -1


def ricerca_binaria(A, n, v):
    sinistra = 0
    destra = n
    while (sinistra <= destra):
        somma = sinistra + destra
        centro = somma / 2 if (somma % 2 == 0) else (somma - 1) / 2
        if (A[centro] > v):
            destra = centro - 1
        elif (A[centro] < v):
            sinistra = centro + 1
        else:
            return centro
    return NON_PRESENTE
